src/widgets/immunizations/immunization-mapper.ts

```diff
@@ -0,0 +1,111 @@
+import { get, groupBy, map, orderBy } from "lodash-es";
+import dayjs from "dayjs";
+
+const mapToImmunizationDoses = immunizationResource => {
+  const immunizationObsUuid = immunizationResource?.resource?.id;
+  const encounterUuid = immunizationResource?.resource?.encounter?.id;
+  const manufacturer = immunizationResource?.resource?.manufacturer;
+  const lotNumber = immunizationResource?.resource?.lotNumber;
+  const protocolApplied =
+    immunizationResource?.resource?.protocolApplied?.length > 0 &&
+    immunizationResource?.resource?.protocolApplied[0];
+  const sequenceLabel = protocolApplied?.series;
+  const sequenceNumber = protocolApplied?.doseNumberPositiveInt;
+  const occurrenceDateTime = dayjs(
+    immunizationResource?.resource?.occurrenceDateTime
+  ).format("YYYY-MM-DD");
+  const expirationDate = dayjs(
+    immunizationResource?.resource?.expirationDate
+  ).format("YYYY-MM-DD");
+  return {
+    encounterUuid: encounterUuid,
+    immunizationObsUuid: immunizationObsUuid,
+    manufacturer: manufacturer,
+    lotNumber: lotNumber,
+    sequenceLabel: sequenceLabel,
+    sequenceNumber: sequenceNumber,
+    occurrenceDateTime: occurrenceDateTime,
+    expirationDate: expirationDate
+  };
+};
+
+export const mapFromFhirImmunizationSearchResults = immunizationSearchResult => {
```

Comment:
Can we type this? What is `immunizationSearchResult`?

Reply:
It's actually the Immunizations for a specific patient. I will try to put a Type around this. Probably `FHIRImmunizationBundle` because that's what it is, A [bundle](https://www.hl7.org/fhir/bundle.html) of all immunizations for patient.